app/scripts/controllers/transactions.js

```diff
@@ -193,7 +189,7 @@ module.exports = class TransactionController extends EventEmitter {
       // wait for a nonce
       nonceLock = await this.nonceTracker.getNonceLock(fromAddress)
       // add nonce to txParams
-      txMeta.txParams.nonce = nonceLock.nextNonce
+      txMeta.txParams.nonce = ethUtil.addHexPrefix(nonceLock.nextNonce.toString(16))
```

Comment:
is there a `ethUtil.intToHex`? if not, there should be

Reply:
i don't think ethereumjs-util should replicate javascript. just do '0x' + int.toString(16) :smile_cat:

Reply:
yeah but we need type validation
```
> `0x${(NaN).toString(16)}`
'0xNaN'
```